pkg/controller/common/webhook/webhook.go

```diff
@@ -101,18 +117,30 @@ func (v *validatingWebhook) Handle(ctx context.Context, req admission.Request) a
 			whlog.Error(err, "decoding old object from webhook request into type (%T)", oldObj)
 			return admission.Errored(http.StatusBadRequest, err).WithWarnings(warnings...)
 		}
-		_, err = obj.ValidateUpdate(oldObj)
-		if err != nil {
-			return admission.Denied(err.Error()).WithWarnings(warnings...)
-		}
+		warnings, err = obj.ValidateUpdate(oldObj)
 	}
 
 	if req.Operation == admissionv1.Delete {
-		_, err = obj.ValidateDelete()
-		if err != nil {
-			return admission.Denied(err.Error())
+		warnings, err = obj.ValidateDelete()
+	}
+	if err != nil {
+		var apiStatus apierrors.APIStatus
+		if errors.As(err, &apiStatus) {
+			return validationResponseFromStatus(false, apiStatus.Status()).WithWarnings(warnings...)
 		}
+		return admission.Denied(err.Error()).WithWarnings(warnings...)
 	}
 
 	return admission.Allowed("").WithWarnings(warnings...)
 }
+
+// validationResponseFromStatus returns a response for admitting a request with provided Status object.
+func validationResponseFromStatus(allowed bool, status metav1.Status) admission.Response {
+	resp := admission.Response{
+		AdmissionResponse: admissionv1.AdmissionResponse{
+			Allowed: allowed,
+			Result:  &status,
+		},
+	}
+	return resp
+}
```

Comment:
nit: unless I'm missing something `allowed` is always false
```suggestion
// denyResponseFromStatus returns a response for admitting a request with provided Status object.
func denyResponseFromStatus(status metav1.Status) admission.Response {
	resp := admission.Response{
		AdmissionResponse: admissionv1.AdmissionResponse{
			Allowed: false,
			Result:  &status,
		},
	}
	return resp
}
```